specs/ServiceProtocol.md

```diff
@@ -100,6 +108,14 @@ MessagePack uses different formats to encode values. Refer to the [MessagePack F
 ```
 - 1 - Message Type, indicating this is a `HandshakeRequest` message.
 - Version - A `Int32` encoding number of the protocol version.
+- ConnectionType - A `Int32` encoding number of the connection type.
+	- 0, Default, it can carry clients, service runtime should always accept this kind of connection.
+	- 1, OnDemand, creating when service requested more connections, it can carry clients, but it may be rejected by service runtime.
+	- 2, Weak, it can not carry clients, but it can send message.
+- MigrationLevel - A `Int32` encoding number indicates if further client connections associated with this server connection could be migrated.
+	- 0, Off, a client connection can not be migrated to another server.
+	- 1, ShutdownOnly, a client connection can be migrated only if the pairing server was shutdown gracefully.
```

Comment:
are we providing too much options to user?

Reply:
Which user, how does anyone get access to this?

Reply:
Our customer could enable this feature by the `ServiceOptions`, like this:
```C#
services.AddSignalR()
                    .AddAzureSignalR(option =>
                    {
                        option.ServerConnectionMigration = ServiceConnectionMigrationLevel.ShutdownOnly
                    });
```

Reply:
Add details for the failed / non mainstream scenarios. E.g. we closed the client on the server and could not open on another server - will the client get disconnected? How many tries before we give up?  Or it took too long to connect to another server - how long does the service try? How much does the service buffer? What happens if the client sends more data than we buffer? What happens if during migration client disconnects - do we still migrate and deliver new messages including client disconnect?